src/app/plan/components/Wizard/HooksStep.tsx

```diff
@@ -0,0 +1,220 @@
+import React, { useEffect, useState, useContext } from 'react';
+import {
+  Table,
+  TableBody,
+  TableHeader,
+} from '@patternfly/react-table';
+import { Spinner } from '@patternfly/react-core/dist/esm/experimental';
+import spacing from '@patternfly/react-styles/css/utilities/Spacing/spacing';
+import { PlusCircleIcon } from '@patternfly/react-icons';
+
+import {
+  Grid,
+  GridItem,
+  Text,
+  TextContent,
+  TextVariants,
+  Bullseye,
+  EmptyState,
+  EmptyStateVariant,
+  EmptyStateBody,
+  Title,
+  Button,
+  EmptyStateIcon,
+  InputGroup,
+  TextInput,
+  Flex,
+  FlexItem,
+  FlexModifiers
+} from '@patternfly/react-core';
+import { SearchIcon } from '@patternfly/react-icons';
+import {
+  AddEditMode,
+} from '../../../common/add_edit_state';
+import HooksFormContainer from './HooksFormContainer';
+
+const classNames = require('classnames');
+const styles = require('./HooksStep.module')
+
+const HooksStep = (props) => {
+  const {
+    updateHookRequest,
+    addHookRequest,
+    isFetchingHookList,
+    migHookList,
+    fetchHooksRequest,
+    hookAddEditStatus,
+    currentPlan,
+    removeHookRequest,
+    watchHookAddEditStatus,
+    isAddHooksOpen,
+    setIsAddHooksOpen
+  } = props;
+
+  const [initialHookValues, setInitialHookValues] = useState({});
```

Comment:
Take it or leave it (we don't really do this in many places, but we probably should): we can add a generic type annotation to `useState` for these complex state values. e.g. `useState<IMigHook>({});`.

Simple `useState` calls with a primitive type in their default value (string, number, boolean) don't need these because TS can infer the type from the initial value and use that to check your usage, but when the default value is an empty object or array we should probably be passing generics like this.

Reply:
+1 done